c7n/resources/redshift.py

```diff
@@ -432,14 +432,15 @@ def set_access(self, c):
 
     def process(self, clusters):
         with self.executor_factory(max_workers=2) as w:
-            futures = []
+            futures = {}
             for c in clusters:
-                futures.append(w.submit(self.set_access, c))
+                futures[w.submit(self.set_access, c)] = c
```

Comment:
Can you change this to use list comprehensions, I would like to standardize the way we are building the process function.

This is how it's done in some of the newer s3 processing:
`{w.submit(self.process_bucket, b, keys): b for b in buckets}`

Reply:
Yeah. I'll do that as well for #1874